clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3130,25 +3145,57 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
       return;
   }
 
-  if (WorkGroupAttr *ExistingAttr = D->getAttr<WorkGroupAttr>()) {
-    ASTContext &Ctx = S.getASTContext();
+  ASTContext &Ctx = S.getASTContext();
+
+  if (!XDimExpr->isValueDependent() && !YDimExpr->isValueDependent() &&
+      !ZDimExpr->isValueDependent()) {
     Optional<llvm::APSInt> XDimVal = XDimExpr->getIntegerConstantExpr(Ctx);
     Optional<llvm::APSInt> YDimVal = YDimExpr->getIntegerConstantExpr(Ctx);
     Optional<llvm::APSInt> ZDimVal = ZDimExpr->getIntegerConstantExpr(Ctx);
-    Optional<llvm::APSInt> ExistingXDimVal = ExistingAttr->getXDimVal(Ctx);
-    Optional<llvm::APSInt> ExistingYDimVal = ExistingAttr->getYDimVal(Ctx);
-    Optional<llvm::APSInt> ExistingZDimVal = ExistingAttr->getZDimVal(Ctx);
 
-    // Compare attribute arguments value and warn for a mismatch.
-    if (ExistingXDimVal != XDimVal || ExistingYDimVal != YDimVal ||
-        ExistingZDimVal != ZDimVal) {
-      S.Diag(AL.getLoc(), diag::warn_duplicate_attribute) << AL;
-      S.Diag(ExistingAttr->getLocation(), diag::note_conflicting_attribute);
+    XDimExpr = checkWorkSizeAttrExpr(S, AL, XDimExpr);
+    YDimExpr = checkWorkSizeAttrExpr(S, AL, YDimExpr);
+    ZDimExpr = checkWorkSizeAttrExpr(S, AL, ZDimExpr);
+
+    if (!XDimExpr || !YDimExpr || !ZDimExpr)
+      return;
+
+    // Skip SEMA if we're in a template, this will be diagnosed later.
+    if (S.getCurLexicalContext()->isDependentContext())
```

Comment:
I'm not certain what this is intending to do? If any of the attribute arguments were template arguments, we wouldn't get into this `if` block at all because of the check for value-dependence above.

Reply:
Yes, it does not make sense here. Sorry about it. This redundant change is removed now